scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -1560,6 +1609,21 @@ private TaskResult loadTaskResult(Session session, TaskId taskId, int resultInde
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private List<TaskResult> loadTaskResultAllAttempts(Session session, TaskId taskId) {
+        DBTaskId dbTaskId = taskId(taskId);
+
+        TaskData task = session.load(TaskData.class, dbTaskId);
+        Query query = session.getNamedQuery("loadTasksResultByTaskAsc").setParameter("task", task);
+
+        List<TaskResultData> resultsData = (List<TaskResultData>) query.list();
+        List<TaskResult> results = new ArrayList<>(resultsData.size());
+        for (TaskResultData resultData : resultsData) {
+            results.add(resultData.toTaskResult(taskId));
+        }
```

Comment:
Myabe:
`
return ((List<TaskResultData>) query.list()).stream().map(resultData -> resultData.toTaskResult(taskId)).collect(Collectors.toList());`

Reply:
ok why not